flink-streaming-connectors/flink-connector-filesystem/src/test/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSinkFaultToleranceITCase.java

```diff
@@ -160,7 +167,15 @@ public void postSubmit() throws Exception {
 				while (line != null) {
 					Matcher matcher = messageRegex.matcher(line);
 					if (matcher.matches()) {
-						numRead++;
+						uniqMessagesRead.add(line);
+
+						// check that in the committed files there are no duplicates
+						if (!file.getPath().toString().endsWith(IN_PROGRESS_SUFFIX) && !file.getPath().toString().endsWith(PENDING_SUFFIX)) {
+							if (!messagesInCommittedFiles.add(line)) {
+								Assert.fail("Duplicate entry in committed bucket.");
```

Comment:
this check isn't truly necessary, is it? We have the total number of messages with `readNumbers.size()`, and the number of unique messages ones in `uniqMessagesRead.size()`. Since we compare both with `NUM_STRINGS` the test can only succeed if no duplicates exist anyway.